src/vector/backends/numpy.py

```diff
@@ -191,14 +191,24 @@ def _setitem(
 
 
 def _getitem(
-    array: VectorNumpy2D | VectorNumpy3D | VectorNumpy4D,
+    array: VectorNumpy2D
+    | VectorNumpy3D
+    | VectorNumpy4D
+    | AzimuthalNumpyRhoPhi
+    | AzimuthalNumpyXY
+    | LongitudinalNumpyZ
+    | LongitudinalNumpyTheta
+    | LongitudinalNumpyEta
+    | TemporalNumpyT
+    | TemporalNumpyTau,
     where: typing.Any,
     is_momentum: bool,
-) -> float | FloatArray:
+) -> float | FloatArray | VectorProtocol | Azimuthal | Longitudinal | Temporal:
```

Comment:
This also implements `__getitem__` for NumPy coordinate types. Updating the type signature allowed me to remove a few `type: ignore[arg-type]` comments in the function body.